hazelcast/src/hazelcast/client/spi/InvocationService.cpp

```diff
@@ -267,7 +275,17 @@ namespace hazelcast {
                 }
 
                 boost::shared_ptr<connection::CallPromise> promise = deRegisterCall(connection, correlationId);
-                if (promise.get() == NULL) {
+                if (NULL == promise.get()) {
```

Comment:
I checked on caller paths for deregister, the access to the callpromisemap for the connection is only in 3 places: registerCall, deregisterCall and cleanReources. None of them removes the entry except the deregisterCall. deregisterCall is called from handleMessage. The only way I can think of in which case the promise is synchronousy removed from the map (so that one of the deregisterCall's getting null response) is if the server sends response with the same correlation id two times and this should be an implementation error.

If you still think, this case may occur in production which may cause too many warning messages, I can revert it, and we can open an issue to investigate further.